lib/subscription/subscription-store.ts

```diff
@@ -1,72 +1,47 @@
 /// <reference path='../../typings/tsd.d.ts' />
 
 import _ = require('lodash');
-import Subscription = require('./subscription');
+import Subscription = require('./subscription-buffer');
+import Map = require('../util/map');
+import interfaces = require('../interface');
 
 export = SubscriptionStore;
 
-class SubscriptionStore<T extends Subscription> {
+class SubscriptionStore extends Map<Subscription> {
```

Comment:
Yeah you could do that too. I tend to prefer not to do that for encapsulation but it may make sense here, I'd have to look at it closer at how it's used.